packages/server/setup/datagen.js

```diff
@@ -150,7 +153,24 @@ class DataGenerator {
     }
   }
 
-  generateDataForPost(post) {
+  async checkIsMature(text) {
+    let isMature = false;
+    const textLengthLimit = 1024;
+    const timeout = 10000;
+
+    for (let i = 0; i < text.length; i += textLengthLimit) {
+      const textModResponse = await callAzureApi(
+        text.substring(i, Math.min(i + textLengthLimit, text.length)),
```

Comment:
If the text is exceptionally long, couldn't this cut the text right where there's a curse word?

Reply:
yep since this is for mock data I didn't make sure to cut exactly on the end of a word, but actual posts are limited to be 1024 chars long